homeassistant/components/hassio/handler.py

```diff
@@ -32,6 +32,19 @@ def _wrapper(*argv, **kwargs):
     return _wrapper
 
 
+def _api_data(funct):
+    """Return a api data."""
+    @asyncio.coroutine
+    def _wrapper(*argv, **kwargs):
+        """Wrap function."""
+        data = yield from funct(*argv, **kwargs)
+        if data and data['result'] == "ok":
+            return data['data']
+        return None
```

Comment:
Wouldn't this make you lose what's inside data if result != ok ?

Reply:
Yes: https://github.com/home-assistant/hassio/blob/dev/API.md#hassio-restful-api there is only a data on 'ok'. On 'error' there is a message. Maybe we can log this message if available.

Reply:
That would be great. I am not a fan of silent errors. I prefer raising.